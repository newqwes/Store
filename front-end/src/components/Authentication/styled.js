import styled, { css } from 'styled-components';
import { get } from 'lodash/fp';

import THEME_VARIANT from '../../constants/themeVariant';

export const FormWrapper = styled.section`
  display: flex;
  margin: auto;
  padding: 20px;

  div {
    margin: 0 auto;
  }
`;

export const FormContent = styled.div(
  ({ themeVariant, isErrorAnimation }) => css`
    padding: 30px 60px;
    background-color: ${get(['theme', 'login', 'background', themeVariant])};
    box-shadow: 0px 0px 20px ${get(['theme', 'login', 'shadow', 'main', themeVariant])};

    animation-name: ${isErrorAnimation && 'error'};
    animation-iteration-count: 1;
    animation-timing-function: ease-in-out;
    animation-duration: 0.8s;
    animation-delay: 0.5s;

    button {
      display: block;
      margin: 15px auto 10px;
      padding: 10px 20px;
      font-size: 16px;
      font-weight: 400;
      border-radius: 0;
      border: 1px solid ${get(['theme', 'login', 'button', 'border', themeVariant])};
      background-color: ${get(['theme', 'login', 'button', 'background', themeVariant])}10;
      color: ${get(['theme', 'login', 'button', 'color', themeVariant])};

      &:active {
        border-radius: 0;
        box-shadow: 0px 0px 20px -5px ${get(['theme', 'login', 'button', 'shadow', themeVariant])};
      }
    }

    a {
      font-size: 15px;
      color: ${get(['theme', 'login', 'link', 'color', themeVariant])};
    }

    @keyframes error {
      50% {
        box-shadow: 0px 0px 20px ${get(['theme', 'login', 'shadow', 'error', themeVariant])};
      }
    }
  `,
);

const getBorderColor = ({ isError, themeVariant }) =>
  get(['theme', 'login', 'input', 'border', isError ? THEME_VARIANT.warning : themeVariant]);

export const InputFieldContent = styled.div(
  ({ themeVariant }) => css`
    label {
      display: block;
      margin: 20px 15px 10px;
      font-size: 16px;
      color: ${get(['theme', 'login', 'label', 'color', themeVariant])};
    }

    input {
      padding: 0.5rem 1rem;
      border: 1px solid ${getBorderColor};
      margin-bottom: 10px;
    }

    span {
      position: absolute;
    }
  `,
);
