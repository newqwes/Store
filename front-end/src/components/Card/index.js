export { default } from './Card.jsx';
export { AddCard } from './AddCard.jsx';
