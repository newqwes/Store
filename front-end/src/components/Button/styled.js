import styled, { css } from 'styled-components';
import { get } from 'lodash/fp';

const ButtonWrapper = styled.button(
  ({ themeVariant }) => css`
    padding: 5px 8px;
    letter-spacing: 1px;
    background-color: ${get(['theme', 'button', 'background', themeVariant])};
    border: 1px solid ${get(['theme', 'button', 'border', themeVariant])};
    border-radius: 5px;

    color: ${get(['theme', 'button', 'color', themeVariant])};
    font-weight: 500;
    outline: none;
    font-size: 12px;
    white-space: nowrap;

    &:hover {
      cursor: pointer;
    }
  `,
);

export default ButtonWrapper;
