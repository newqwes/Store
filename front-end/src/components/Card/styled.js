import { get } from 'lodash/fp';
import styled, { css } from 'styled-components';

import LabelWrapper from '../Label/styled';

export const CardWrapper = styled.div`
  display: flex;
  flex-basis: 19%;
  flex-direction: column;
  align-content: space-between;
  position: relative;
  margin: 28px 7px;
  padding: 10px 20px;
  transition: 0.3s ease;
  border-radius: 20px;
`;

export const Description = styled(LabelWrapper)`
  opacity: 0.7;
`;

export const CardContent = styled.div(
  ({ themeVariant }) => css`
    display: flex;
    flex-wrap: wrap;
    justify-content: flex-start;
    line-height: 18px;

    &:last-child {
      margin-top: auto;
    }

    button {
      margin-left: 16px;
      letter-spacing: 0;
      border: none;
      font-weight: normal;
      border-radius: 20px;
      padding: 8px 20px;
      font-size: 16px;
      line-height: 24px;
      background-color: ${get(['theme', 'card', 'button', 'background', themeVariant])};
      color: ${get(['theme', 'card', 'button', 'color', themeVariant])};
      transition: 0.3s;

      :hover {
        background-color: ${get(['theme', 'card', 'button', 'hover', themeVariant])};
      }

      :active {
        background-color: ${get(['theme', 'card', 'button', 'background', themeVariant])};
      }
    }

    > * {
      display: flex;
      margin: 5px;
    }
  `,
);
