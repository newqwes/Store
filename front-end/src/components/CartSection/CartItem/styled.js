import styled, { css } from 'styled-components';
import { get } from 'lodash/fp';

export const CartItemContent = styled.div(
  ({ themeVariant }) => css`
    margin-bottom: 15px;
    background-color: white;
    padding: 10px;
    border-radius: 5px;
    box-shadow: 0 0 10px -8px ${get(['theme', 'cart', 'shadow', themeVariant])};
  `,
);

export const CartPhotoContent = styled.div`
  flex-basis: 15%;
  background: no-repeat center/80% url(${props => props.backgroundImage});
  height: 80px;
  align-self: center;
`;

export const CartDescriptionContent = styled.div`
  flex-basis: 65%;

  span {
    display: block;
    margin-top: 10px;
  }
`;

export const CartPriceContent = styled.div`
  flex-basis: 20%;

  span {
    min-width: 40px;
    text-align: center;
  }

  div {
    margin-top: 10px;
  }

  button {
    font-size: 16px;
    width: 30px;
    height: 30px;
  }
`;
