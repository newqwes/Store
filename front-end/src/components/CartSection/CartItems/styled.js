import styled from 'styled-components';

import SECTION_WIDTH from '../../../constants/sectionWidth';

const CartItemsWrapper = styled.div`
  width: 100%;
  margin: 0 auto;
  max-width: ${SECTION_WIDTH.midle}px;
`;

export default CartItemsWrapper;
