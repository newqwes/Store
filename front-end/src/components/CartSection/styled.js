import styled from 'styled-components';

import SECTION_WIDTH from '../../constants/sectionWidth';

export const CartSectionWrapper = styled.section`
  margin: auto 0;
  padding: 20px;
`;

export const CartSectionContent = styled.div`
  margin: 0 25%;
  padding: 20px 0;
`;

export const TotalPriceContent = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: flex-end;
  margin: 0 25%;
  max-width: ${SECTION_WIDTH.midle}px;

  button {
    font-size: 15px;
    margin: 50px 0;
    padding: 10px;
  }
`;
