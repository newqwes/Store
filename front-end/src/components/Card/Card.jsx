import React from 'react';
import find from 'lodash/find';
import PropTypes from 'prop-types';
import { getOr, first, toNumber } from 'lodash/fp';

import { productType } from '../../propType';

import THEME_VARIANT from '../../constants/themeVariant';
import FONT_SIZE from '../../constants/fontSize';
import FONT_WEIGHT from '../../constants/fontWeight';
import TEXT_ALIGN from '../../constants/textAlign';

import EditProductPopupContainer from '../../containers/EditProductPopupContainer';

import Image from '../Image';
import Label from '../Label';
import Flex from '../Flex';
import Select from '../Select';
import Button from '../Button';
import Pen from '../Icons/Pen';
import ValueFormatter from '../ValueFormatter';

import { CardWrapper, CardContent, Description } from './styled';

class Card extends React.Component {
  static propTypes = {
    item: productType.isRequired,
    themeVariant: PropTypes.string,
    updateProduct: PropTypes.func.isRequired,
    addToCart: PropTypes.func.isRequired,
    editMode: PropTypes.bool.isRequired,
  };

  static defaultProps = {
    themeVariant: THEME_VARIANT.default,
  };

  getOptions = () => getOr([], ['item', 'options'], this.props);

  state = {
    option: first(this.getOptions()),
    showPopup: false,
  };

  findByWeight = value => {
    const options = this.getOptions();
    return find(options, ['weight', toNumber(value)]);
  };

  handleChange = ({ target }) => {
    const option = this.findByWeight(target.value);
    this.setState({ option });
  };

  addToCart = () => {
    const { addToCart, item } = this.props;
    const { option } = this.state;

    return addToCart({ ...item, options: option });
  };

  togglePopup = () => this.setState({ showPopup: !this.state.showPopup });

  render() {
    const {
      editMode,
      themeVariant,
      updateProduct,
      item: { photoUrl, name, description, currencySign, options, unitSign },
    } = this.props;

    const { option, showPopup } = this.state;

    return (
      <CardWrapper themeVariant={themeVariant}>
        {editMode && <Pen onClick={this.togglePopup} />}
        <Image src={photoUrl} alt={name} />
        <CardContent themeVariant={themeVariant}>
          <Label text={name} fontSize={FONT_SIZE.medium} fontWeight={FONT_WEIGHT.normal} />
          <Description fontSize={FONT_SIZE.least} fontWeight={FONT_WEIGHT.normal}>
            {description}
          </Description>
          <Flex>
            <ValueFormatter
              value={option.price}
              textAlign={TEXT_ALIGN.center}
              fontWeight={FONT_WEIGHT.normal}
              postfix={currencySign}
            />
            <Select options={options} onChange={this.handleChange} unitSign={unitSign} />
            <Button text='Выбрать' onClick={this.addToCart} />
          </Flex>
        </CardContent>
        {showPopup && (
          <EditProductPopupContainer
            initialValues={this.props.item}
            closePopup={this.togglePopup}
            submitPopup={updateProduct}
          />
        )}
      </CardWrapper>
    );
  }
}

export default Card;
