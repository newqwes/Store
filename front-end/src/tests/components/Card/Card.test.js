import React from 'react';

import { mountWithTheme } from '../../helpers';
import { mockProductItem } from '../../helpers/mocks';

import { CardWrapper, CardContent, Description } from '../../../components/Card/styled';

import Card from '../../../components/Card';

const setUp = props =>
  mountWithTheme(
    <Card
      item={mockProductItem}
      userAdmin={true}
      addToCart={() => {}}
      updateProduct={() => {}}
      {...props}
    />,
  );

describe('snapshot Card', () => {
  test('snapshot', () => {
    const wrapper = setUp();

    expect(wrapper).toMatchSnapshot();
  });
});

describe('Card component', () => {
  const mockAddToCart = jest.fn();
  const mockupdateProduct = jest.fn();

  const wrapper = setUp({
    addToCart: mockAddToCart,
    updateProduct: mockupdateProduct,
  });

  it(`should render Card with following children:
  - 1 CardWrapper,
  - 1 CardContent,
  - 1 Flex,
  - 1 Label,
  - 1 Description,
  - 1 ValueFormatter,
  - 1 Select,
  - 1 Button,
  - 0 Pen`, () => {
    const cardWrapper = wrapper.find(CardWrapper);
    expect(cardWrapper.length).toBe(1);

    const cardContent = wrapper.find(CardContent);
    expect(cardContent.length).toBe(1);

    const description = wrapper.find(Description);
    expect(description.length).toBe(1);

    const flexs = wrapper.find('Flex');
    expect(flexs.length).toBe(1);

    const labels = wrapper.find('Label');
    expect(labels.length).toBe(1);

    const buttons = wrapper.find('Button');
    expect(buttons.length).toBe(1);

    const valueFormatter = wrapper.find('ValueFormatter');
    expect(valueFormatter.length).toBe(1);

    const select = wrapper.find('Select');
    expect(select.length).toBe(1);

    const pen = wrapper.find('Pen');
    expect(pen.length).toBe(0);
  });

  it('should be call addToCart when click button', () => {
    const buttons = wrapper.find('Button');
    buttons.simulate('click');

    expect(mockAddToCart).toHaveBeenCalledTimes(1);
  });
});
