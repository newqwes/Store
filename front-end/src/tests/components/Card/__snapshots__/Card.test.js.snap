// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot Card snapshot 1`] = `
<div
  className="sc-gsTCUz jiEHUK"
>
  <div
    className="sc-idOhPF ivwdKp"
  >
    <img
      alt="Pizza"
      src="https://image.com"
    />
  </div>
  <div
    className="sc-hKgILt fIBcqK"
  >
    <span
      className="sc-bdfBwQ fJgcr"
      fontSize={20}
      fontWeight={400}
    >
      Pizza
    </span>
    <span
      className="sc-bdfBwQ sc-dlfnbm loYUik cSdHaW"
      fontSize={12}
      fontWeight={400}
    >
      none description
    </span>
    <div
      className="sc-eCssSg irwUHJ"
      direction="row"
    >
      <span
        className="sc-dmlrTW kmkMLJ"
        fontSize={20}
        fontWeight={400}
      >
        24
        руб.
      </span>
      <select
        className="sc-dIUggk kzOnOa"
        onChange={[Function]}
      >
        <option
          key="250"
          value={250}
        >
          250
           
          г
        </option>
      </select>
      <button
        className="sc-fubCfw gXwbFj"
        onClick={[Function]}
      >
        Выбрать
      </button>
    </div>
  </div>
</div>
`;
